continuous_integration.md

```diff
@@ -6,147 +6,209 @@ done with the corresponding commands `deno test`, `deno lint` and `deno fmt`. In
-is configured to trigger a workflow run on push and pull request events. Note
-that in the example the `latest` version of `ubuntu` image is used, but you
-could specify an exact version for added stability in a production pipeline,
-such as `ubuntu-20.04`.
+```yaml
+name: Build
 
-To expand the workflow you can add any of the `deno` CLI commands that you might
-need. The code below shows how to check the formatting, lint the code, run the
-tests and generate a test coverage report, all as part of a `build` job:
+on: push
 
-```yaml
 jobs:
   build:
     runs-on: ubuntu-20.04
     steps:
-      - name: Set up Actions
-        uses: actions/checkout@v2
-
-      - name: Set up Deno
-        uses: denoland/setup-deno@v1.0.0
+      - uses: actions/checkout@v2
+      - uses: denoland/setup-deno@v1.0.0
         with:
-          deno-version: v1.x
-
-      - name: Check formatting
-        run: deno fmt --check
+          deno-version: v1.x # Run with latest stable Deno.
+```
 
-      - name: Analyze code
-        run: deno lint
+To expand the workflow just add any of the `deno` subcommands that you might
+need:
 
-      - name: Run unit and integration tests
-        run: deno test -A --coverage=cov --doc
+```yaml
+      # Check if the code is formatted according to Deno's default
+      # formatting conventions.
+      - run: deno fmt --check
 
-      - name: Generate coverage report
-        run: deno coverage --lcov cov > cov.lcov
-```
+      # Scan the code for syntax errors and style issues. If
+      # you want to use a custom linter configuration you can add a configuration file with --config <myconfig>
+      - run: deno lint
 
-Let's go over the steps one by one.
+      # Run all test files in the repository and collect code coverage. The example
+      # runs with all permissions, but it is recommended to run with the minimal permissions your program needs (for example --allow-read).
+      - run: deno test --allow-all --coverage cov/
 
-```yaml
-- name: Check formatting
-  run: deno fmt --check
+      # This generates a report from the collected coverage in `deno test --coverage`. It is
+      # stored as a .lcov file which integrates well with services such as Codecov, Coveralls and Travis CI.
+      - run: deno coverage --lcov cov/ > cov.lcov
 ```
 
-This simply checks if the project code is formatted according to Deno's default
-formatting conventions.
+### Cross-platform workflows
+
+As a Deno module maintainer, you probably want to know that your code works on
+all of the major operating systems in use today: Linux, MacOS and Windows. A
+cross-platform workflow can be achieved by running a matrix of parallel jobs,
+each one running the build on a different OS:
 
 ```yaml
-- name: Analyze code
-  run: deno lint
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-20.04, macos-11, windows-2019 ]
+    steps:
+      # Build goes here.
```

Comment:
I suggest to replace this comment with at least one step (probably `deno test` would be best)

Reply:
Done!